### PR TITLE
Next gen docs for 1.9

### PR DESCRIPTION
**Pull request by NigelBreslaw**
WIP PR to update the docs for 1.9 using Astro Starlight

### CODE REVIEW

**Comment by ogoffart** [comment]
I wonder if it would be possible to rebase that to have a cleaner history.  @hunger: could you help with that?

**Comment by tronical** [comment]
I’ll be back Wednesday, will help try out and review then (here or master if merged by then).

File: tools/docsnapper/LICENSES/LicenseRef-Slint-Software-3.0.md
Comment (tronical) [inline tools/docsnapper/LICENSES/LicenseRef-Slint-Software-3.0.md:1]: This file has a path to Nigel's local file system. Seems wrong :-)

File: tools/docsnapper/headless.rs [hunk 1]
```diff
@@ -0,0 +1,187 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
```
Comment (tronical) [inline tools/docsnapper/headless.rs:4]: Double license header here.

File: tools/docsnapper/main.rs [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+    #[arg(long, value_name = "component name", action)]
+    component: Option<String>,
+}
+
+fn main() -> Result<()> {
+    let args = Cli::parse();
+
+    let project_root = find_project_root(&args.docs_folder)?;
+
+    eprintln!("project_root is {project_root:?}");
```
Comment (tronical) [inline tools/docsnapper/main.rs:73]: Stray debug output?

**Review by tronical — APPROVED** [review]

**Pull request merged**